### PR TITLE
Resolve Avatar size shortcode attribute issue in donor wall shortcode and adds support for avatar size in donor wall block

### LINKED ISSUE #5442: Avatar Size shortcode attribute should affect actual avatar size and other Donor Wall avatar improvements
## User Story

As a admin, I want to control the avatar size on the donor wall so that it can fit my design goals.

Additionally, in the Donor Wall block there is no option to change the avatar size. 

Finally, when the Gravatar appears it's not the correct resolution and appears grainy / pixelated because it's coming from the API at 80x80px and we're smushing it to 60x60px. We should ensure that it's always pulled at 2x the size (if displaying at 60px the image should be 120px) for the highest possible resolution. So if I'm an admin and set the `avatar_size` to 120 then we would pull from Gravatar a 240px resolution image. 

## Details

Currently the `avatar_size` parameter is not working. This stems from the work that was done to validate the Gravatar by @ravinderk - in doing that PR we left out testing the resize functionality, which is now broken. 

## Expected Behavior

The `avatar_size` parameter should work as expected alongside the new gravatar validation functionality.

### Shortcode

1. Insert a donor_wall shortcode with the `avatar_size` parameter set to `120`. Example: `[donor_wall avatar_size="120"]`.
2. View on the frontend and notice the 

### Block

1. Insert the "Donor Wall" block
2. Notice there's no option to change the avatar size

### Resolution

1. View the donor wall on any device and notice the image looks grainy / pixelated. 

## Acceptance Criteria

-   [x] `avatar_size` works in the shortcode
-   [x] The Donor Wall block supports changing the size of the avatar
-	[x] The donor's gravatar appears high resolution and quality
- 	[x] The donor wall is properly responsive with a larger or smaller sized avatar

### PR DESCRIPTION
**Pull request by DevinWalker**
Resolves #5442 

## Description

We find out that donor wall shortcode has support for avatar size attribute but value of this attribute hardccoded in donor wall template. @DevinWalker fixed that. I added support to donor wall Gutenberg block for avatar size. Avatar loads in 2x size of avatar size shortcode attribute value.

image

## Testing

- Create a donor wall block and check if you can configure `Avatar size`
- Edit avatar size in donor wall shortcode

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+-   Resolve Avatar size shortcode attribute issue in donor wall shortcode and adds support for avatar size in donor wall block (#5443)
```
Comment (JasonTheAdams) [inline CHANGELOG.md:48]: Please move this into the `Unreleased` section

**Comment by DevinWalker** [comment]
This isn't working at all for me: https://www.loom.com/share/0f700085774542e0b2193fd6d0e55901

File: CHANGELOG.md (hunk 1 above)
Comment (DevinWalker) [reply CHANGELOG.md:50]: This has been moved

Comment (ravinderk) [reply CHANGELOG.md:50]: @DevinWalker thank you for moving that. I misplace when resolving conflict.

Comment (ravinderk) [reply CHANGELOG.md:52]: @DevinWalker I noticed that it was not placed under a section (`Added, Fixed, etc.`). I moved it to the `Unreleased-->Added` section.

**Comment by ravinderk** [comment]
@DevinWalker I did not get you before. I understand requirement and i am working on fix.

**Comment by ravinderk** [comment]
@DevinWalker To make donor wall avatar responsive i divided container to 40(image)-60(text content). This means that donor image size depends upon available 40% space in container.

**Desktop**
image

**Vertical Phone**
image

**Horizontal phone**
image


## Need help
1. Anonymous donor svg image has space around image that why you will notice that border is not showing correctly. Can you provide me anonymous donor svg image?

image


2. As you can see screenshot, donor name initial feature is working fine for me. I think that some how `_give_donor_billing_last_name` contain space as first character which cause of only one letter is showing in name initial. Can you debug this on your end why `give_get_name_initial` is returning wrong string?
In case you are not able to debug this then provide me your local site, so i can reproduce this at my end to fix it.

**Comment by DevinWalker** [comment]
> @DevinWalker To make donor wall avatar responsive i divided container to 40(image)-60(text content). This means that donor image size depends upon available 40% space in container.

Ok that should be fine, I'll test it out as well to see it in action. Did you blur your avatar image out in the screenshots above?

> Anonymous donor svg image has space around image that why you will notice that border is not showing correctly. Can you provide me anonymous donor svg image?

I'll send you the SVG on Slack

> As you can see screenshot, donor name initial feature is working fine for me. I think that some how _give_donor_billing_last_name contain space as first character which cause of only one letter is showing in name initial. Can you debug this on your end why give_get_name_initial is returning wrong string? In case you are not able to debug this then provide me your local site, so i can reproduce this at my end to fix it.

I'm not sure why this would be happening, I just spun up my local again a few days ago but I'll check again.

**Comment by ravinderk** [comment]
@DevinWalker

1. Yes, I blur my image in the above screenshots.
2. New SVG dimension looks good but I can notice a few gaps. I also noticed that the previous image was SVG with transparent background.
image

**Comment by DevinWalker** [comment]
This is good to merge. @ravinderk I noticed one issue with the gravatar not displaying properly in the Gutenberg editor but displaying properly on the frontend. That's pretty minor so if it's quick to fix that would be great but if not go ahead and merge it: https://www.loom.com/share/8a870defb3ea4d9b8075729d2794e4cd

**Comment by ravinderk** [comment]
@DevinWalker I look into the gravatar loading issue in Gutenberg block and I do not have a quick for this. I will create a separate issue for this.

**Pull request merged**